tests/light/functional_tests/filterx/test_filterx.py

```diff
@@ -1534,3 +1534,130 @@ def test_unset_empties(config, syslog_ng):
     assert file_true.get_stats()["processed"] == 1
     assert "processed" not in file_false.get_stats()
     assert file_true.read_log() == "[{},{},[],[]]\n"
+
+
+def test_null_coalesce_happy_path(config, syslog_ng):
```

Comment:
For the E2E tests generally:

It is the cleanest to test only one functionality in one TC, but there is a performance impact, which I think should be considered. All these TCs start syslog-ng, which has a significant overhead, we can quickly end up with minutes long E2E test runs this way.

I didn't think through if any of these TCs could be merged sanely, but this "feels" like that we should handle these cases in fever TCs for this simple feature to save test time.